processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -1677,30 +1691,67 @@ public void getFromByteBuffer(ByteBuffer buffer, int initialOffset, int dimValId
       @Override
       public int compare(ByteBuffer lhsBuffer, ByteBuffer rhsBuffer, int lhsPosition, int rhsPosition)
       {
-        return Double.compare(
-            lhsBuffer.getDouble(lhsPosition + keyBufferPosition),
-            rhsBuffer.getDouble(rhsPosition + keyBufferPosition)
-        );
+        return bufferComparator.compare(lhsBuffer, rhsBuffer, lhsPosition, rhsPosition);
       }
     }
+  }
 
-    private class LimitPushDownDoubleRowBasedKeySerdeHelper extends DoubleRowBasedKeySerdeHelper
-    {
-      final StringComparator cmp;
-
-      public LimitPushDownDoubleRowBasedKeySerdeHelper(int keyBufferPosition, StringComparator cmp)
-      {
-        super(keyBufferPosition);
-        this.cmp = cmp;
+  private static int compareDimsInBuffersForNullFudgeTimestamp(
+      List<RowBasedKeySerdeHelper> serdeHelpers,
+      int dimCount,
+      ByteBuffer lhsBuffer,
+      ByteBuffer rhsBuffer,
+      int lhsPosition,
+      int rhsPosition
+  )
+  {
+    for (int i = 0; i < dimCount; i++) {
+      final int cmp = serdeHelpers.get(i).compare(
+          lhsBuffer,
+          rhsBuffer,
+          lhsPosition + Longs.BYTES,
+          rhsPosition + Longs.BYTES
+      );
+      if (cmp != 0) {
+        return cmp;
       }
+    }
 
-      @Override
-      public int compare(ByteBuffer lhsBuffer, ByteBuffer rhsBuffer, int lhsPosition, int rhsPosition)
-      {
-        double lhs = lhsBuffer.getDouble(lhsPosition + keyBufferPosition);
-        double rhs = rhsBuffer.getDouble(rhsPosition + keyBufferPosition);
-        return cmp.compare(String.valueOf(lhs), String.valueOf(rhs));
+    return 0;
+  }
+
+  private static int compareDimsInBuffersForNullFudgeTimestampForPushDown(
+      List<RowBasedKeySerdeHelper> serdeHelpers,
+      List<Boolean> needsReverses,
```

Comment:
Couldn't use `BitSet`?

Reply:
Hmm, I guess this method was accidentally moved. The logic around 'limitPushDown for groupBy' is quite complicated. I would like to do refactoring in another pr if needed.